src/admin/units/units.component.tsx

```diff
@@ -1,7 +1,199 @@
-import { Typography } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import ClearIcon from '@mui/icons-material/Clear';
+import DeleteIcon from '@mui/icons-material/Delete';
+import {
+  Box,
+  Button,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  TableRow,
+} from '@mui/material';
+import { useUnits } from '../../api/units';
+import {
+  MRT_ColumnDef,
+  MaterialReactTable,
+  useMaterialReactTable,
+} from 'material-react-table';
+import { Unit } from '../../app.types';
+import { usePreservedTableState } from '../../common/preservedTableState.component';
+import { MRT_Localization_EN } from 'material-react-table/locales/en';
+import React from 'react';
+import { formatDateTimeStrings, getPageHeightCalc } from '../../utils';
+import UnitsDialog from './unitsDialog.component.tsx';
+import DeleteUnitDialog from './deleteUnitsDialog.component.tsx';
 
 function Units() {
-  return <Typography>Work in progress</Typography>;
+  const { data: unitData, isLoading: unitDataLoading } = useUnits();
+
+  const tableHeight = getPageHeightCalc('50px + 110px + 48px');
+
+  const [deleteUnitDialog, setDeleteUnitDialog] =
+    React.useState<boolean>(false);
+
+  const [selectedUnit, setSelectedUnit] = React.useState<Unit | undefined>(
+    undefined
+  );
+
+  const columns = React.useMemo<MRT_ColumnDef<Unit>[]>(() => {
+    return [
+      {
+        header: 'Value',
+        accessorFn: (row) => row.value,
+        id: 'value',
+        Cell: ({ row }) => row.original.value,
+      },
+      {
+        header: 'Last modified',
+        accessorFn: (row) => new Date(row.modified_time),
+        id: 'modified_time',
+        filterVariant: 'datetime-range',
+        size: 350,
+        enableGrouping: false,
+        Cell: ({ row }) =>
+          row.original.modified_time &&
+          formatDateTimeStrings(row.original.modified_time, true),
+      },
+      {
+        header: 'Created',
+        accessorFn: (row) => new Date(row.created_time),
+        id: 'created_time',
+        filterVariant: 'datetime-range',
+        size: 350,
+        enableGrouping: false,
+        enableHiding: true,
+        Cell: ({ row }) =>
+          formatDateTimeStrings(row.original.created_time, true),
+      },
+    ];
+  }, []);
+
+  const noResultsTxt =
+    'No results found: Try adding a Unit by using the Add Unit button';
+
+  const { preservedState, onPreservedStatesChange } = usePreservedTableState({
+    initialState: {
+      pagination: { pageSize: 15, pageIndex: 0 },
+    },
+    storeInUrl: true,
+  });
+
+  const table = useMaterialReactTable({
+    columns: columns,
+    data: unitData ?? [],
+    // Features
+    enableColumnOrdering: true,
+    enableColumnResizing: true,
```

Comment:
```suggestion
```

Reply:
Can you also remove this from the usages statues table as well please. This would allow the columns to fill up the whole screen